dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -0,0 +1,241 @@
+            UpdateCompositionGradientEllipseRadius();
+            UpdateCompositionGradientOriginOffset();
+            UpdateCompositionGradientStops();
+            UpdateCompositionGradientMappingMode();
+            UpdateCompositionInterpolationSpace();
+        }
+        else
+        {
+            // If CompositionRadialGradientBrush isn't available then render using the FallbackColor.
+            m_brush = compositor.CreateColorBrush();
+            UpdateFallbackBrush();
+        }
+
+        CompositionBrush(m_brush);
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientEllipseCenter()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        compositionGradientBrush.EllipseCenter(winrt::float2(EllipseCenter().X, EllipseCenter().Y));
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientEllipseRadius()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        compositionGradientBrush.EllipseRadius(winrt::float2(EllipseRadius().X, EllipseRadius().Y));
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientMappingMode()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        switch (MappingMode())
+        {
+        case winrt::BrushMappingMode::Absolute:
+            compositionGradientBrush.MappingMode(winrt::Windows::UI::Composition::CompositionMappingMode::Absolute);
+            break;
+
+        case winrt::BrushMappingMode::RelativeToBoundingBox:
+            compositionGradientBrush.MappingMode(winrt::Windows::UI::Composition::CompositionMappingMode::Relative);
+            break;
+        }
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientOriginOffset()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        compositionGradientBrush.GradientOriginOffset(winrt::float2(GradientOriginOffset().X, GradientOriginOffset().Y));
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientStops()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        auto compositor = winrt::Window::Current().Compositor();
+
+        compositionGradientBrush.ColorStops().Clear();
+
+        for (uint32_t i = 0; i < m_gradientStops.Size(); i++)
```

Comment:
we could use a range based for loop here.
for(const auto& gradientStop: m_gradientStops)
{
    ...
}